addons/auth_passkey/models/key.py

```diff
@@ -0,0 +1,65 @@
+import base64
+
+from odoo import api, fields, models, _
+from odoo.addons.base.models.res_users import check_identity
+
+
+class PassKey(models.Model):
+    _name = "auth.passkey.key"
+    _description = "Passkeys"
+    _order = "id desc"
+
+    name = fields.Char(required=True)
+    credential_identifier = fields.Char(required=True, groups='base.group_system')
+    public_key = fields.Char(required=True, groups='base.group_system')
+    sign_count = fields.Integer(default=0, groups='base.group_system')
+
+    _sql_constraints = [
+        ('unique_identifier', 'UNIQUE(credential_identifier)', 'The credential identifier should be unique.'),
+    ]
+
+    def _get_user_by_credential_id(self, identifier):
+        identifier = base64.urlsafe_b64decode(identifier).hex()
+        result = self.sudo().search([("credential_identifier", "=", identifier)])
+        return result
+
+    @check_identity
+    @api.model
+    def action_new_passkey(self, key):
+        self.sudo().create({
+            'name': key['name'],
+            'credential_identifier': base64.urlsafe_b64decode(key['credential_identifier']).hex(),
+            'public_key': key['public_key'],
+        })
```

Comment:
`sudo().create` is an attack vector pattern.

For instance, the user will be able to control `sign_count` through this method, despite the `groups='base.group_system'` added in the field, calling this method with 
```py
env['auth.passkey.key'].with_context(default_sign_count=1337).action_new_passkey()
```
I guess you won't do a lot with this. And, as you are "forcing" the other values name, credential_identifier and sign_count, and there is no other fields, other `default_` keys won't work, and it's relatively fine.
But, the day you won't "force" all values, or we will add new fields in the model, then it could become critical.

It will be even more critical the day there is a one2many or many2many, as then you will be able to pass commands as sudo, for instance `[(0, 0, {'foo': 'bar', 'bar': 'xyz'})]`, and pass arbitrary fields on the co-model as sudo.

I was wondering how they were avoiding this in the API keys similar method.
Using raw SQL:
https://github.com/odoo/odoo/blob/4bec7151c1f594678c3576de1255b4609c3e0c88/odoo/addons/base/models/res_users.py#L2146-L2166

Another argument why to go for the extra security and extract `credential_identifier` and `public_key` out of the ORM. And use raw SQL at read/write of these fields.

Also `_generate` is made private on the API key model itself, and is called from the wizard model `res.users.apikeys.description`, then form `make_key` protected by a `check_identity`